### PR TITLE
Get a new token when the token expires

### PR DESCRIPTION
**Pull request by fangyi-zhou**
Not sure how to test this change, advice appreciated

### CODE REVIEW

**Comment by Rigellute** [comment]
I'm going to leave the app open for 1 hour, and then try to play a track 😄 .

Hilarious way to test.

Will get back to you.

File: src/main.rs [hunk 1]
```diff
@@ -142,6 +153,18 @@ fn main() -> Result<(), failure::Error> {
...
 
+                if Instant::now() > token_expiry {
+                    // refresh token
+                    if let Some(new_token_info) = get_token(&mut oauth) {
+                        let (new_spotify, new_token_expiry) = get_spotify(new_token_info);
+                        spotify = new_spotify;
+                        token_expiry = new_token_expiry;
+                        app.spotify = Some(spotify);
+                    } else {
+                        // TODO: Handle error
```
Comment (Rigellute) [inline src/main.rs]: For now, on error let's just exit like this

```rust
println!("\nFailed to refresh authentication");
break;
```

The new line in the print is so the log will show in the console.

**Comment by Rigellute** [comment]
I've been testing your fork and it's looking good I think!

If you add the `break` when auth fails, this can be merged 👍

**Comment by fangyi-zhou** [comment]
Rebased and added changelog

**Comment by stevensonmt** [comment]
I'm curious how spotify-tui works with tokens. I am using it with spotifyd, and I thought spotifyd was requesting the API token. My solution to the token expiring was to have spotifyd.service restart every 3500 seconds. Does this PR alleviate the need for restarting the spotifyd.service?

**Comment by Rigellute** [comment]
Yes, this PR fixes the auth token expiring issue.

`spotify-tui` uses the Spotify Web API to control playback on devices (such as spotifyd), and so needs it's own auth separate to spotifyd.

You can think of `spotify-tui` as a "controller" for spotify devices.

**Pull request merged**